src/test/java/org/ibp/api/java/impl/middleware/common/validator/AttributeValidatorTest.java

```diff
@@ -58,11 +80,185 @@ public void testForValidProgramId() throws MiddlewareQueryException {
 
 		final Attribute attribute = new Attribute();
 		attribute.setAid(attributeById);
-		List<Attribute> attributeLists = Lists.newArrayList(attribute);
+		final List<Attribute> attributeLists = Lists.newArrayList(attribute);
 		Mockito.doReturn(attributeLists).when(this.germplasmDataManager).getAttributeByIds(Lists.newArrayList(attributeById));
 
 		this.attributeValidator.validateAttributeIds(bindingResult, Lists.newArrayList(String.valueOf(attributeById)));
 
 		Assert.assertFalse(bindingResult.hasErrors());
 	}
+
+	@Test(expected = ApiRequestValidationException.class)
+	public void testValidateAttributeType() {
```

Comment:
In our first versions for the validator tests we were using the expected but it is much better to catch the exception so you can confirm that the error code is the right one. 
Please take a look at `GermplasmImportRequestDtoValidatorTest#testValidateBeforeSaving_ThrowsException_WhenRequestIsNull`
Also notice that your function name does not clearly express what you want to test. We not have any standard in BMS for naming our tests but see the names I have been using in GermplasmImportRequestDtoValidatorTest so you can add the condition that is tested to the function names
Please notice that each test should test only one condition. In some funtions you are testing more than one condition
This comment applies to all functions in this class so I dont repeat the same in all places